### ISSUE TITLE
please provide a switch to close auto detect cargo.toml or connect network

### ISSUE THREAD

**Issue by Silentdoer** [intellij-rust/intellij-rust#6234]
the function is for offline environment, i want refresh by self.

**Comment by Undin** [comment]
@Silentdoer Did you try `Watch Cargo.toml` option in `Preferences | Languages & Frameworks | Rust | Cargo`?
Also, there is `Offline mode` option to pass `--offline`

**Comment by Silentdoer** [comment]
Yes, I have checked the Offline mode in ...|Rust|Cargo, but when i use idea open a rust project (have no Cargo.lock file), it is always connect network, tips: Updating crates.io index

**Comment by Silentdoer** [comment]
i have a common Cargo.lock file, and pack some crates, and copy thease file to offline machine, i want create Cargo.lock and paste the config by myself, then i refresh Cargo, cargo will not connect network.

**Comment by njasm** [comment]
I have a similar issue that I still didn't found a workaround.

With latest version of the plugin and CLion I open a rust project, and I don't want the plugin to generate a Cargo.lock if its not present.

I'm struggling into trying to stop this  assumption from the plugin, that I do want to generate a cargo.lock just cos its missing.

**Comment by Undin** [comment]
> With latest version of the plugin and CLion I open a rust project, and I don't want the plugin to generate a Cargo.lock if its not present.

@njasm I suppose it's impossible for now because the plugin uses `cargo metadata` command to retrieve info about project and the command generates `Cargo.lock` file

BTW why do you want not to generate `Cargo.lock`?

**Comment by njasm** [comment]
It's just a safeguard to not push a file I wasn't supposed to be changing. 
I have the bad habit of `git add .` but with the plugin I always need to make a mental note to not add the lock file. (if it's not already in the `.ignore`)

Some times the dependencies are defined not until the patch version in the `Cargo.toml` 
ex: `dependency = "0.2"`.  (I understand that this practice have a lot of room for discussion) 

for instance if a new patch version of a dependency shows up on `crates.io` I might accidentally be pushing a new state for the dependencies graph of the project.
